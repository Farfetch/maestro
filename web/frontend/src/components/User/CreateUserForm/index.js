import CreateUserForm from "./CreateUserForm";

export default CreateUserForm;
