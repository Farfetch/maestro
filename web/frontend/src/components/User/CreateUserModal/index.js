import CreateUserModal from "./CreateUserModal";

export default CreateUserModal;
