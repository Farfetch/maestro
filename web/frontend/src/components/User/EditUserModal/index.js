import EditUserModal from "./EditUserModal";

export default EditUserModal;
