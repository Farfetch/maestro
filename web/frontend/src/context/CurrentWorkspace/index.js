export {
  CurrentWorkspaceContext,
  CurrentWorkspaceContextProvider
} from "./CurrentWorkspace";
