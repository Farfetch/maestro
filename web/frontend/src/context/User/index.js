export { UserContext, UserContextProvider } from "./UserContext";
