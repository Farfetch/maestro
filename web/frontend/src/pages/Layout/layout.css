@import "~antd/dist/reset.css";

/* https://ant.design/components/layout/ */
:root {
  --breakpoint-xs: 480px;
  --breakpoint-sm: 576px;
  --breakpoint-lg: 768px;
  --breakpoint-lg: 992px;
  --breakpoint-xl: 1200px;
  --breakpoint-xxl: 1600px;
}

body {
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen",
    "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue",
    sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;

  margin: 0;
}

code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, "Courier New",
    monospace;
}

main {
  padding: 1em;
  max-width: var(--breakpoint-xl);
  margin: 0 auto;
}

@media (max-width: var(--breakpoint-lg)) {
  main {
    max-width: none;
  }
}

/* Custom Error Message */
.ant-message-notice-content {
  display: inline-block;
  /* background: #efecec; */
  border-radius: 5px;
  white-space: pre-wrap;
  color: rgb(44, 42, 42);
}

.ant-message {
    top: 20px;
}

/* Page Header */
.ant-page-header-heading {
  align-items: center;
}


.react-resizable {
  position: relative;
  background-clip: padding-box;
}

.react-resizable-handle {
  position: absolute;
  right: -5px;
  bottom: 0;
  z-index: 1;
  width: 10px;
  height: 100%;
  cursor: col-resize;
}

.truncate-option {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

