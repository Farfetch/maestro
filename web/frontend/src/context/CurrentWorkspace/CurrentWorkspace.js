import { createContext, useContext, useState } from "react";

import { UserContext } from "../User";

export const CurrentWorkspaceContext = createContext(null);

export const CurrentWorkspaceContextProvider = ({ children }) => {
  const { workspaces } = useContext(UserContext);
  const [currentWorkspace, setCurrentWorkspace] = useState(workspaces[0]);

  return (
    <CurrentWorkspaceContext.Provider
      value={{ currentWorkspace, setCurrentWorkspace }}
    >
      {children}
    </CurrentWorkspaceContext.Provider>
  );
};
