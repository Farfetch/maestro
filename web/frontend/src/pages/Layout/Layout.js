import "./layout.css";

import { Outlet } from "react-router-dom";

import Header from "../../components/Header";

function Layout() {
  return (
    <main>
      <Header />
      <div>
        <Outlet />
      </div>
    </main>
  );
}

export default Layout;
