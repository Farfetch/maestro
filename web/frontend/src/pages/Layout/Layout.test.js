import { render } from "@testing-library/react";
import React from "react";
import { MemoryRouter } from "react-router-dom";

import { CurrentWorkspaceContext } from "../../context/CurrentWorkspace";
import { UserContext } from "../../context/User";
import Layout from "./Layout";

describe("pages/Layout", () => {
  test(`should render Layout component`, async () => {
    const user = {
      id: "user-id",
      name: "Test Maestro",
      email: "test@maestro.net"
    };
    const workspaces = [
      {
        id: "workspace-1",
        name: "workspace-1"
      },
      {
        id: "workspace-2",
        name: "workspace-2"
      }
    ];

    const rendered = render(
      <UserContext.Provider value={{ user, workspaces }}>
        <CurrentWorkspaceContext.Provider
          value={{ currentWorkspace: workspaces[0] }}
        >
          <MemoryRouter initialEntries={["/"]}>
            <Layout />
          </MemoryRouter>
        </CurrentWorkspaceContext.Provider>
      </UserContext.Provider>
    );

    const badgeComponent = rendered.container;
    expect(badgeComponent.outerHTML).toMatchSnapshot();
  });
});
