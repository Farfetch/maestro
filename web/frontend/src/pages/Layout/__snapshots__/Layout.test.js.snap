// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/Layout should render Layout component 1`] = `"<div><main><header><div class=\\"ant-row\\"><div style=\\"flex-basis: auto;\\" class=\\"ant-col\\"><div class=\\"ant-space ant-space-horizontal ant-space-align-center\\"><div class=\\"ant-space-item\\" style=\\"margin-right: 24px;\\"><a href=\\"/\\"><svg width=\\"200\\" height=\\"48\\" viewBox=\\"0 0 317 73\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"M42.12 58L42 35.74L31.2 53.86H25.92L15.18 36.34V58H4.2V16H13.98L28.74 40.24L43.14 16H52.92L53.04 58H42.12ZM89.5861 49.84H71.8261L68.5261 58H56.4061L74.9461 16H86.6461L105.246 58H92.8861L89.5861 49.84ZM86.1061 41.08L80.7061 27.64L75.3061 41.08H86.1061ZM142.334 48.82V58H108.614V16H141.554V25.18H120.374V32.26H139.034V41.14H120.374V48.82H142.334ZM163.687 58.84C160.287 58.84 156.987 58.42 153.787 57.58C150.587 56.74 147.987 55.62 145.987 54.22L149.887 45.46C151.767 46.7 153.947 47.7 156.427 48.46C158.907 49.22 161.347 49.6 163.747 49.6C168.307 49.6 170.587 48.46 170.587 46.18C170.587 44.98 169.927 44.1 168.607 43.54C167.327 42.94 165.247 42.32 162.367 41.68C159.207 41 156.567 40.28 154.447 39.52C152.327 38.72 150.507 37.46 148.987 35.74C147.467 34.02 146.707 31.7 146.707 28.78C146.707 26.22 147.407 23.92 148.807 21.88C150.207 19.8 152.287 18.16 155.047 16.96C157.847 15.76 161.267 15.16 165.307 15.16C168.067 15.16 170.787 15.48 173.467 16.12C176.147 16.72 178.507 17.62 180.547 18.82L176.887 27.64C172.887 25.48 169.007 24.4 165.247 24.4C162.887 24.4 161.167 24.76 160.087 25.48C159.007 26.16 158.467 27.06 158.467 28.18C158.467 29.3 159.107 30.14 160.387 30.7C161.667 31.26 163.727 31.84 166.567 32.44C169.767 33.12 172.407 33.86 174.487 34.66C176.607 35.42 178.427 36.66 179.947 38.38C181.507 40.06 182.287 42.36 182.287 45.28C182.287 47.8 181.587 50.08 180.187 52.12C178.787 54.16 176.687 55.8 173.887 57.04C171.087 58.24 167.687 58.84 163.687 58.84ZM196.128 25.42H183.228V16H220.848V25.42H208.008V58H196.128V25.42ZM243.634 46.84H237.154V58H225.274V16H244.474C248.274 16 251.574 16.64 254.374 17.92C257.174 19.16 259.334 20.96 260.854 23.32C262.374 25.64 263.134 28.38 263.134 31.54C263.134 34.58 262.414 37.24 260.974 39.52C259.574 41.76 257.554 43.52 254.914 44.8L263.974 58H251.254L243.634 46.84ZM251.134 31.54C251.134 29.58 250.514 28.06 249.274 26.98C248.034 25.9 246.194 25.36 243.754 25.36H237.154V37.66H243.754C246.194 37.66 248.034 37.14 249.274 36.1C250.514 35.02 251.134 33.5 251.134 31.54ZM290.868 58.84C286.428 58.84 282.428 57.9 278.868 56.02C275.348 54.14 272.568 51.54 270.528 48.22C268.528 44.9 267.528 41.16 267.528 37C267.528 32.84 268.528 29.1 270.528 25.78C272.568 22.46 275.348 19.86 278.868 17.98C282.428 16.1 286.428 15.16 290.868 15.16C295.308 15.16 299.288 16.1 302.808 17.98C306.368 19.86 309.148 22.46 311.148 25.78C313.188 29.1 314.208 32.84 314.208 37C314.208 41.16 313.188 44.9 311.148 48.22C309.148 51.54 306.368 54.14 302.808 56.02C299.288 57.9 295.308 58.84 290.868 58.84ZM290.868 49C292.988 49 294.908 48.5 296.628 47.5C298.348 46.5 299.708 45.1 300.708 43.3C301.708 41.46 302.208 39.36 302.208 37C302.208 34.64 301.708 32.56 300.708 30.76C299.708 28.92 298.348 27.5 296.628 26.5C294.908 25.5 292.988 25 290.868 25C288.748 25 286.828 25.5 285.108 26.5C283.388 27.5 282.028 28.92 281.028 30.76C280.028 32.56 279.528 34.64 279.528 37C279.528 39.36 280.028 41.46 281.028 43.3C282.028 45.1 283.388 46.5 285.108 47.5C286.828 48.5 288.748 49 290.868 49Z\\" fill=\\"#CB0000\\"></path><path d=\\"M309.6 37C309.6 46.7001 301.293 54.6 291 54.6C280.707 54.6 272.4 46.7001 272.4 37C272.4 27.2999 280.707 19.4 291 19.4C301.293 19.4 309.6 27.2999 309.6 37Z\\" stroke=\\"#E8CACA\\" stroke-width=\\"0.8\\"></path><path d=\\"M231 55V42.5M231 42.5V20H240.346L245.679 20.3423C247.304 20.4467 248.9 20.8274 250.397 21.4682L250.771 21.6284C252.118 22.2047 253.348 23.0206 254.403 24.0366L254.444 24.0759C255.082 24.69 255.647 25.3749 256.129 26.1175L256.611 26.8602C257.518 28.2568 258 29.886 258 31.551V31.551C258 32.5102 257.84 33.4628 257.526 34.3693L257.224 35.2415C256.71 36.7261 255.852 38.0681 254.721 39.1578L254.688 39.1889C254.129 39.7276 253.506 40.1964 252.834 40.585V40.585C251.787 41.1895 250.636 41.5909 249.441 41.768L246.381 42.2213C245.129 42.4069 243.864 42.5 242.598 42.5H236.712H231Z\\" stroke=\\"#E8CACA\\" stroke-width=\\"0.8\\"></path><path d=\\"M247 42.5L255.5 55\\" stroke=\\"#E8CACA\\" stroke-width=\\"0.8\\"></path><path d=\\"M202 54.5V21M202 21H217M202 21H187\\" stroke=\\"#E8CACA\\" stroke-width=\\"0.8\\"></path><path d=\\"M137.5 53.5H113.5V37M113.5 37V20.5H137.5M113.5 37H133\\" stroke=\\"#E8CACA\\" stroke-width=\\"0.8\\"></path><path d=\\"M68 45.5H93.5\\" stroke=\\"#E8CACA\\"></path><path d=\\"M9.5 55V20L28.5 48.5L47.9969 20V55\\" stroke=\\"#E8CACA\\" stroke-width=\\"0.8\\"></path><path d=\\"M63.5 55L80.5 20L98.5 55\\" stroke=\\"#E8CACA\\" stroke-width=\\"0.8\\"></path><path d=\\"M175 22L172.542 20.9075C171.186 20.3049 169.759 19.8766 168.296 19.6327L168.165 19.6108C165.75 19.2084 163.277 19.3335 160.915 19.9776L160.211 20.1697C159.406 20.3891 158.63 20.7018 157.898 21.1012L156.522 21.8517C154.607 22.8964 153.253 24.7361 152.825 26.8753V26.8753C152.611 27.9451 152.637 29.049 152.902 30.1074L152.925 30.1999C153.592 32.8661 155.613 34.984 158.246 35.7737L159 36L163.02 36.804C165.332 37.2665 167.584 37.9927 169.731 38.9685L171.148 39.6129C171.715 39.8705 172.26 40.1735 172.778 40.5189L174.747 41.8312C175.875 42.5833 176.682 43.7288 177.011 45.0441V45.0441C177.327 46.3062 177.18 47.6396 176.598 48.8032L176.245 49.5098C175.754 50.4916 175.056 51.3551 174.199 52.0408L174.12 52.1037C173.379 52.6966 172.539 53.1536 171.639 53.4537L170.644 53.7854C169.224 54.2587 167.737 54.5 166.24 54.5H163L162.202 54.4335C158.743 54.1453 155.324 53.4973 152 52.5V52.5\\" stroke=\\"#E8CACA\\" stroke-width=\\"0.8\\"></path></svg></a></div><div class=\\"ant-space-item\\" style=\\"margin-right: 24px;\\"><a href=\\"/history\\">History</a></div><div class=\\"ant-space-item\\" style=\\"margin-right: 24px;\\"><a href=\\"/tests\\">Tests</a></div><div class=\\"ant-space-item\\"><a href=\\"/agents\\">Agents</a></div></div></div><div style=\\"margin-left: auto; display: flex;\\" class=\\"ant-col\\"><div class=\\"ant-space ant-space-horizontal ant-space-align-center\\"><div class=\\"ant-space-item\\" style=\\"margin-right: 8px;\\"><div class=\\"ant-select ant-select-single ant-select-show-arrow\\" style=\\"width: 240px;\\"><div class=\\"ant-select-selector\\"><span class=\\"ant-select-selection-search\\"><input type=\\"search\\" autocomplete=\\"off\\" class=\\"ant-select-selection-search-input\\" role=\\"combobox\\" aria-haspopup=\\"listbox\\" aria-owns=\\"rc_select_TEST_OR_SSR_list\\" aria-autocomplete=\\"list\\" aria-controls=\\"rc_select_TEST_OR_SSR_list\\" aria-activedescendant=\\"rc_select_TEST_OR_SSR_list_0\\" readonly=\\"\\" unselectable=\\"on\\" style=\\"opacity: 0;\\" value=\\"\\" id=\\"rc_select_TEST_OR_SSR\\"></span><span class=\\"ant-select-selection-item\\" title=\\"workspace-1\\">workspace-1</span></div><span class=\\"ant-select-arrow\\" style=\\"user-select: none;\\" unselectable=\\"on\\" aria-hidden=\\"true\\"><span role=\\"img\\" aria-label=\\"down\\" class=\\"anticon anticon-down ant-select-suffix\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" data-icon=\\"down\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z\\"></path></svg></span></span></div></div><div class=\\"ant-space-item\\"><div style=\\"cursor: pointer;\\" class=\\"ant-dropdown-trigger\\"><span style=\\"background-color: rgb(250, 140, 22);\\" class=\\"ant-avatar ant-avatar-circle\\"><span class=\\"ant-avatar-string\\" style=\\"transform: scale(1) translateX(-50%);\\">TM</span></span></div></div></div></div></div></header><div></div></main></div>"`;
